src/modules/emotes.js

```diff
@@ -146,83 +146,49 @@ function EmoteStore() {
 		// Hash of emote set to forced channel.
 		var forcedSetsToChannels = {
 			// Globals.
-			0: '~global',
+			0: 'Global Twitch',
 			// Bubble emotes.
-			33: 'turbo',
+			33: 'Twitch Turbo',
 			// Monkey emotes.
-			42: 'turbo',
+			42: 'Twitch Turbo',
 			// Hidden turbo emotes.
-			457: 'turbo',
-			793: 'turbo',
-			19151: 'twitch_prime',
-			19194: 'twitch_prime'
-
+			457: 'Twitch Turbo',
+			793: 'Twitch Turbo',
+			19151: 'Twitch Prime',
+			19194: 'Twitch Prime'
 		};
 
-		logger.debug('Initializing emote set change listener.');
-
-		twitchApi.getEmotes(function (emoteSets) {
-			logger.debug('Parsing emote sets.');
-
-			Object.keys(emoteSets).forEach(function (set) {
-				var emotes = emoteSets[set];
-				set = Number(set);
-				emotes.forEach(function (emote) {
-					// Set some required info.
-					emote.url = '//static-cdn.jtvnw.net/emoticons/v1/' + emote.id + '/1.0';
-					emote.text = getEmoteFromRegEx(emote.code);
-					emote.set = set;
-
-					// Hardcode the channels of certain sets.
-					if (forcedSetsToChannels[set]) {
-						emote.channel = forcedSetsToChannels[set];
-					}
-
-					var instance = new Emote(emote);
-
-					// Save the emote for use later.
-					nativeEmotes[emote.text] = instance;
-				});
-			});
-
-			logger.debug('Loading subscription data.');
+		var setsToChannels = {};
 
-			// Get active subscriptions to find the channels.
-			twitchApi.getTickets(function (tickets) {
-				// Instances from each channel to preload channel data.
-				var deferredChannelGets = {};
+		logger.debug('Getting emote sets and user emotes');
 
-				logger.debug('Tickets loaded from the API.', tickets);
-				tickets.forEach(function (ticket) {
-					var product = ticket.product;
-					var channel = product.owner_name || product.short_name;
-
-					// Get subscriptions with emotes only.
-					if (!product.emoticons || !product.emoticons.length) {
-						return;
-					}
-					
-					// Set the channel on the emotes.
-					product.emoticons.forEach(function (emote) {
-						var instance = nativeEmotes[getEmoteFromRegEx(emote.regex)];
-						instance.setChannelName(channel);
-
-						// Save instance for later, but only one instance per channel.
-						if (!deferredChannelGets[channel]) {
-							deferredChannelGets[channel] = instance;
+		twitchApi.getEmoteSets(function (setsToChannels) {
+			twitchApi.getUserEmotes(function (emotesBySet) {
+				Object.keys(emotesBySet).forEach(function (set) {
+					var emotes = emotesBySet[set];
+					set = +set;
+					emotes.forEach(function (emote) {
+						// Set some required info.
+						emote.url = '//static-cdn.jtvnw.net/emoticons/v1/' + emote.id + '/2.0';
```

Comment:
Why was this changed from `.../1.0` to `.../2.0`? The one used in chat is `.../1.0`. I see that `.../2.0` is higher quality but some of the emotes end up stretched (old emotes). Wouldn't it be better to match the quality shown in chat?

Reply:
I see what Twitch is actually doing (and it's new to me). They are actually defining a [`srcset`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-srcset).

In addition to `src` being the `.../1.0`, they are also defining `srcset` as `.../1.0 1.0x, .../2.0 2.0x, .../3.0 3.0x` which picks the best image to serve based on resolution/zoom.

I'll definitely add formal support for this to get the best quality possible.